### PR TITLE
Fix Issue that When two or more Modules exists in same packages, Only one module created

### PR DESCRIPTION
**Pull request by aerain**
## Issue
The title says it all.

I guess reason of moduleMap existence is handling ComponentScan annotations. 
but When moduleMap is initialized. some of modules were lost.

I think #11 is related to this issue 🤔

### CODE REVIEW

File: compiler/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/scanner/KoinScanner.kt [hunk 1]
```diff
@@ -46,14 +46,14 @@ class KoinMetaDataScanner(
...
         defaultModule: KoinMetaData.Module
-    ): Map<String, KoinMetaData.Module> {
+    ): ModuleMap {
 
         logger.logging("scan modules ...")
         // class modules
         moduleMap = resolver.getSymbolsWithAnnotation(Module::class.qualifiedName!!)
             .filter { it is KSClassDeclaration && it.validate() }
             .map { moduleMetadataScanner.createClassModule(it) }
-            .toMap()
```
Comment (aerain) [inline compiler/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/scanner/KoinScanner.kt:56]: root cause

Comment (arnaudgiuliani) [reply compiler/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/scanner/KoinScanner.kt:56]: ok, interesting 🤔

**Comment by arnaudgiuliani** [comment]
I don't understand here the point. Sample app with 2 modules in same package, are generated:

image

**Comment by aerain** [comment]
maybe i mistitled. 🤔

Remove `@ComponentScan` annotation from them.

Otherwise, CoffeeAppModule will be registered in key `org.koin.example.coffee` and, CoffeeTesterModule will be registered in key `org.koin.example.test`

for now, that two modules are not conflicted in sample.

this issue is occurred when condition if
* two modules are located in other packages. but has same ComponentScan value.
* two modules are located in same packages. (two modules shouldn't have ComponentScan annotation with different value.)

**Comment by aerain** [comment]
check this module map key generator.
https://github.com/InsertKoinIO/koin-annotations/blob/main/compiler/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/scanner/ModuleScanner.kt#L51

```kt
 val key = if (componentScan?.packageName?.isNotEmpty() == true) componentScan.packageName else modulePackage
```

**Comment by arnaudgiuliani** [comment]
> two modules are located in other packages. but has same ComponentScan value.
Shouldn't work and then compiler should warn that we try to rescan a package

> two modules are located in same packages.
Should have a different component scan, or can't work

What do you think? 🤔

**Comment by aerain** [comment]
> > two modules are located in other packages. but has same ComponentScan value.
> 
> Shouldn't work and then compiler should warn that we try to rescan a package

Same opinion.


> > two modules are located in same packages.
> 
> Should have a different component scan, or can't work

Why is ComponentScan necessary? I think it's strong constraint to use.


sample use-case of second case.

foo/config/aws/DynamoDbModule.kt
```kt
package foo.config.aws

@Module
class DynamoDbModule {
    @Single
    fun dynamoDb(): DynamoDBClient = ...
}
```

foo/config/aws/S3Module.kt
```kt
package foo.config.aws

@Module
class S3Module {
    @Single
    fun s3Client(): S3Client = ...
}
```

I think this use case should be allowed.

**Comment by aerain** [comment]
I added Module validator for this case. please check.



> > two modules are located in other packages. but has same ComponentScan value.
>
> Shouldn't work and then compiler should warn that we try to rescan a package

**Comment by arnaudgiuliani** [comment]
let me cleanup everything. I will close your PR here. Once it will be deployed on 1.0 stable, you will tell me 👍

**Pull request closed without merging**